### DISCUSSION TITLE
Plans for additional theme.json sections

### DISCUSSION THREAD

**Discussion by luisherranz** [WordPress/gutenberg#35474] [Developer Experience]
Right now, the `theme.json` has 4 main sections:

- Settings (global, per block, or per element):
  - Define Presets
  - Define Custom CSS properties
  - Show/hide customization options from the user
  - Define Layout sizes
- Styles (global, per block, or per element)
- Custom templates
- Template parts

I wonder if there are other sections on the roadmap or ideas for the `theme.json` file that have not been implemented yet.

**Comment by mtias** [comment]
There's a lot of ideas in different stages of development included in https://github.com/WordPress/gutenberg/issues/20331. Notably, the idea is that theme.json files could be composed at different layers of the theme hierarchy so it can match a template (`archive`, `home`) or an individual page (`page-123`). The idea is that template parts could also become theme.json objects in their own right, meaning you could overwrite or specify settings for a sidebar (allow a different set of color presets than the rest of the site).

**Reply by luisherranz** [reply]
Thanks a lot for the response Matías, and sorry for my late answer.

> Notably, the idea is that theme.json files could be composed at different layers of the theme hierarchy so it can match a template

Sounds great 🙂

> The idea is that template parts could also become theme.json objects in their own right, meaning you could overwrite or specify settings for a sidebar

What about patterns? _Maybe with a UI toggle to bypass it and use the installed `theme.json`._

Also, a question related to what you shared in your [comment in the theme.json v2](https://github.com/WordPress/gutenberg/issues/34349#issuecomment-939036550) issue regarding this:

> A dark theme could also be conceptualized under that model easily.

Any ideas about how that would work? A `theme-dark.json`? If so:
- What about multiple templates?
  - `single.json` -> `single-dark.json`
  - `page-123.json` -> `page-123-dark.json`
  - ...
- Any plans to make that work for arbitrary variables, not only dark mode? 
  - `theme-blue.json`
  - `theme-red.json`
  - ...

> There's a lot of ideas in different stages of development included in #20331.

That's quite a long read taking into account the number of issues linked, so I'll get back to you once I have time to go over that 😄
